### PR TITLE
AppArmor, SELinux, Pi detection, Bionic support

### LINKED ISSUE #211: Add support for Ubuntu 18.04 (Bionic)
We should expand our x86 support to include Ubuntu 18.04 as well, as it is the latest LTS release.

### LINKED ISSUE #224: Check AppArmor and SELinux status in Snap
Currently we check this by running AppArmor and SELinux executables which is forbidden in Snap. Other methods are available, for example we can use python3-apparmor and python3-selinux packages (only available as deb).

Try using these packages inside Snap and see if they are allowed to do their job (at least read-only). If not, devise another way of getting AppArmor/SELinux status.

### LINKED ISSUE #226: Improve Pi detection
`/proc/cpuinfo` in Ubuntu Core and Server does not contain "Hardware" and "Serial" fields. Instead some other files should be queried.

To get device model we can read `/proc/device-tree/model`. It contains data like `Raspberry Pi 3 Model B Plus Rev 1.3\0` (notice the null byte at the end). Serial number is at `/proc/device-tree/serial-number`. Also `/proc/device-tree/compatible` contains info in such format: `raspberrypi,3-model-b-plus brcm,bcm2837`.

### PR DESCRIPTION
**Pull request by a-martynovich**
Closes #226 
Closes #224 
Closes #211 
**Requires** wottsecurity/api#426

### CODE REVIEW

File: agent/security_helper.py [hunk 1]
```diff
@@ -83,38 +83,45 @@ def is_app_armor_enabled():
     Returns a True/False if AppArmor is enabled.
     """
     try:
-        from sh import aa_status
+        import LibAppArmor
+        return LibAppArmor.aa_is_enabled() == 1
```
Comment (rptrchv) [inline agent/security_helper.py]: pls move this out of `try`

File: agent/security_helper.py [hunk 2]
```diff
@@ -83,38 +83,45 @@ def is_app_armor_enabled():
...
     except ImportError:
-        return False
+        try:
+            from sh import aa_status
+        except ImportError:
```
Comment (rptrchv) [inline agent/security_helper.py]: pls comment what all execution branches mean

File: agent/security_helper.py [hunk 3]
```diff
@@ -83,38 +83,45 @@ def is_app_armor_enabled():
...
+            return False
 
-    # Returns 0 if enabled and 1 if disable
-    get_aa_status = aa_status(['--enabled'], _ok_code=[0, 1]).exit_code
-    if get_aa_status == 1:
-        return False
-    return True
+        # Returns 0 if enabled and 1 if disabled
+        return aa_status(['--enabled'], _ok_code=[0, 1]).exit_code != 1
```
Comment (rptrchv) [inline agent/security_helper.py]: it's not clear why didn't you put **all** possible return codes of `aa_status` to `_ok_code`

File: agent/security_helper.py [hunk 4]
```diff
@@ -83,38 +83,45 @@ def is_app_armor_enabled():
...
         {'enabled': False, 'mode': 'enforcing'}
     """
-    selinux_enabled = None
+    selinux_enabled = False
     selinux_mode = None
 
     try:
-        from sh import sestatus
+        import selinux
+        if selinux.is_selinux_enabled() == 1:
```
Comment (rptrchv) [inline agent/security_helper.py]: pls move this branch out of `try`

File: agent/security_helper.py (hunk 3 above)
Comment (a-martynovich) [reply agent/security_helper.py]: fair enough, needs fixing

File: agent/security_helper.py (hunk 2 above)
Comment (rptrchv) [reply agent/security_helper.py]: nope. e.g. it's not clear what does it mean if `LibAppArmor` isn't available for import

Comment (a-martynovich) [reply agent/security_helper.py]: ok I'll add a comment

**Comment by rptrchv** [comment]
@a-martynovich and pls fix the Codacy warning

**Comment by a-martynovich** [comment]
@rptrchv Does it make sense to you? To me it doesn't. If I remove `self` from there it will be impossible to call this method on an instance (`is_file() takes 0 positional arguments but 1 was given`).
image

**Comment by rptrchv** [comment]
> @rptrchv Does it make sense to you? To me it doesn't.

I've already put this exact pylint check to the skip-list in my recent task. Don't do anything re this. I wrote my comment here without looking into the error

**Pull request merged**